server/option.go

```diff
@@ -335,6 +340,15 @@ func WithGRPCUnknownServiceHandler(f func(ctx context.Context, methodName string
 	}}
 }
 
+func WithUnknownMethodHandler(f unknowns.UnknownMethodService) Option {
+	return Option{F: func(o *internal_server.Options, di *utils.Slice) {
+		di.Push(fmt.Sprintf("WithUnknownMethodHandler(%+v)", utils.GetFuncName(f)))
+		o.RemoteOpt.UnknownMethodHandler = f
+		remote.PutPayloadCode(serviceinfo.Thrift, unknown.NewUnknownCodec(thrift.NewThriftCodec()))
```

Comment:
`remote.PutPayloadCode` is already called in `internal/server/option.go`, no need to be called again here.

Reply:
The original Thrift and pb need to be replaced

Reply:
you should not set codec by remote.PutPayloadCode, it is a global configuration.
set it this way `o.RemoteOpt.PayloadCodec = xxx`

Reply:
func NewUnknownServiceCodec(payloadCodecs map[serviceinfo.PayloadCodec]PayloadCodec)